core/logger/default.go

```diff
@@ -0,0 +1,160 @@
+package logger
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"go.uber.org/zap"
+)
+
+var (
+	// Default logger for use throughout the project.
+	// All the package-level functions are calling Default.
+	Default *Logger
+)
+
+func init() {
+	err := zap.RegisterSink("pretty", prettyConsoleSink(os.Stderr))
+	if err != nil {
+		log.Fatalf("failed to register pretty printer %+v", err)
+	}
+	err = registerOSSinks()
+	if err != nil {
+		log.Fatalf("failed to register os specific sinks %+v", err)
+	}
+
+	zl, err := zap.NewProduction()
+	if err != nil {
+		log.Fatal(err)
+	}
+	SetLogger(&Logger{
+		SugaredLogger: zl.Sugar(),
+	})
+}
+
+// SetLogger sets the internal logger to the given input.
+//
+// DEPRECATED this method is deprecated because it leads to race conditions.
+// Instead, you should fork the logger.Default instance to create a new logger
+// for your module.
+// Eg: logger.Default.Named("<my-package-name>")
+func SetLogger(newLogger *Logger) {
```

Comment:
With the intent to phase it out over time?

Reply:
That's the plan, yeah! 
Tbh, I don't have a plan to do that.